updater.py

```diff
@@ -34,33 +36,84 @@ def buildFromSource(tag):
     # Move binary
     os.system("mv gitea "+settings.gtfile)
 
+def is_tool(name):
+    ##Check whether `name` is on PATH and marked as executable. 
+
+    # from whichcraft import which
+    from shutil import which
+
+    return which(name) is not None
+
+if not is_tool("xz"):
+	print ("missing dependency: xz")
+	quit()
+
 # Version from gitea site
 current_version = requests.get(settings.gtsite).json()['version']
-
+print ("current_version =", current_version)
 # Get version tag from github and remove first char (v)
 github_version_tag = requests.get(settings.gtgithubapiurl).json()['tag_name']
-
+print ("github_version_tag =", github_version_tag)
 # Get version from version tag
 github_version = github_version_tag[1:]
-
 # Check if there is a new version
 if github_version > current_version:
 
+
     # Stop systemd service
-    os.system("systemctl stop gitea.service")
+    print ("new version available, stopping service")
+    #os.system("systemctl stop gitea.service")
 
     # Should the new version be build from source?
     if settings.build_from_source:
 
         buildFromSource(github_version_tag)
 
     else:
-
         # Set download url
-        gtdownload = 'https://github.com/go-gitea/gitea/releases/download/'+github_version_tag+'/gitea-'+github_version+'-'+settings.gtsystem
+        ## main file
+        gtdownload = 'https://github.com/go-gitea/gitea/releases/download/'+github_version_tag+'/gitea-'+github_version+'-'+settings.gtsystem+'.xz'
+        print (gtdownload)
+        ## sha256 file
+        shadownload = 'https://github.com/go-gitea/gitea/releases/download/'+github_version_tag+'/gitea-'+github_version+'-'+settings.gtsystem+'.xz.sha256'
+        print (shadownload)
 
         # Download file
-        download(gtdownload, settings.gtfile)
+
+        ## downloading sha
+        print ("downloading sha256 hashsum")
+        download(shadownload, settings.tmpdir+'gitea.xz.sha256')
+        ## downloading xz
+        print ("downloading", github_version_tag+'gitea.xz')
+        tmpxz = settings.tmpdir+'gitea-'+github_version+'-'+settings.gtsystem+'.xz'
+        download(gtdownload, tmpxz)
+
+        # doing sha256 sum
+        os.chdir(settings.tmpdir)
+        #sha_value = os.system("sha256sum -c gitea.xz.sha256 > /dev/null")
+
+        if os.system("sha256sum -c gitea.xz.sha256 > /dev/null") == 0:
```

Comment:
My goal is to put the operating logic into functions and only call the functions in the updater.py file.

I would be very happy if you could put the logic of the sha-check into a function of the functions.py file ;-)

Reply:
With the current content of does 2 lines, i don't think you should move it to a function.

It can be more cumbersome to debug if you do the `chdir ` via functions, since it is need for sha256sum, but can affect other programs using root-dir as reference.

with only  part of a line, there is nothing to save by putting it in a function.